src/init.cpp

```diff
@@ -1267,6 +1266,19 @@ bool AppInitMain(NodeContext& node, interfaces::BlockAndHeaderTipInfo* tip_info)
         }
     }
 
+    for (const std::string& socket_addr : args.GetArgs("-torcontrol")) {
+        std::string host_out;
+        uint16_t port_out{0};
+        size_t colon = socket_addr.find_last_of(':');
+        bool fHaveColon = colon != socket_addr.npos;
```

Comment:
I believe here. You might additionally check the count of `:` characters in `socket_addr` and return an error if they are more than one.